snapcraft/application.py

```diff
@@ -238,26 +209,13 @@ def _run_inner(self) -> int:
     @override
     def _enable_craft_parts_features(self) -> None:
         """Enable partitions if components are defined."""
-        if self._snapcraft_yaml_data and self._snapcraft_yaml_data.get("components"):
-            craft_parts.Features(enable_partitions=True)
-
-    @override
-    def _setup_partitions(self, yaml_data: dict[str, Any]) -> list[str] | None:
-        components = models.ComponentProject.unmarshal(yaml_data)
-        if components.components is None:
-            return None
-
-        return components.get_partitions()
+        try:
```

Comment:
(minor) maybe put this repeated try/except in a `def _get_project_raw() -> dict[str,Any] | None`
then you can walrus it as `if project := self._get_project_raw():`

Reply:
Good idea